frontend/src/metabase/actions/containers/ActionParametersInputForm/ActionParametersInputForm.tsx

```diff
@@ -92,69 +97,32 @@ function ActionParametersInputForm({
     fetchInitialValues,
   ]);
 
-  const fieldSettings = useMemo(
-    () =>
-      action.visualization_settings?.fields ??
-      // if there are no field settings, we generate them from the parameters and field metadata
-      generateFieldSettingsFromParameters(
-        missingParameters,
-        dashcard?.card?.result_metadata as unknown as Field[],
-      ),
```

Comment:
FWIW, you wouldn't find this `result_metadata` twist repeated anywhere else again as we're no longer doing this on `master` (this is essentially a dead codepath)